### PR TITLE
nixos.programs.neovim: add vimdiffAlias option

### PR DESCRIPTION
**Pull request by blackheaven**
## Description of changes

I have added an option to have `vimdiff` alias to NeoViM

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
why do that instead of changing vimdiff into  nvim -d or vim -d ?  Let's not add more code for something this trivial

**Comment by blackheaven** [comment]
I'm not sure to understand, can you rephrase it please?

**Comment by teto** [comment]
neovim made the choice to remove vimdiff to make things simpler, I see no point of reintroducing the complexity in nixpkgs. What for ? nvim -d is the same number of characters, syntax doesn't change. you can add yourself a vimdiff executable or `nvim -d`

**Comment by blackheaven** [comment]
I'm not sure to understand.

I thought the `viAlias` and `vimAlias` was to keep the compatibility when migrating to NeoViM, if so it lacks a proper `vimdiff` link to be complete (which can be used to configure `git`).

I'm not seeing my contribution as a cause of huge complexity tbh.

**Comment by teto** [comment]
> I thought the viAlias and vimAlias was to keep the compatibility when migrating to NeoViM, if so it lacks a proper vimdiff link to be complete (which can be used to configure git).

I disagree with the viAlias and vimAlias settings, nvim != vi or vim. They predate my maintainership and are kept for backwards compatibility. 

Can't you use `nvimdiff` that is supported by any recent git (just run `git difftool --tool-help`). 

Any code has a maintainance cost.
